recaptcha/recaptcha-enterprise/src/main/kotlin/com/pingidentity/recaptcha/enterprise/ReCaptchaEnterpriseCallback.kt

```diff
@@ -0,0 +1,224 @@
+import kotlinx.serialization.json.jsonPrimitive
+
+/**
+ * Configuration class for ReCaptcha Enterprise verification settings.
+ *
+ * This DSL configuration class allows customization of ReCaptcha Enterprise
+ * verification parameters including site key, action type, timeout, and
+ * the client provider implementation.
+ *
+ * Example usage:
+ * ```kotlin
+ * callback.verify {
+ *     recaptchaAction = RecaptchaAction.LOGIN
+ *     timeoutInMills = 15000L
+ *     customPayload = buildJsonObject {
+ *         put("metadata", "value")
+ *     }
+ * }
+ * ```
+ *
+ * @property recaptchaAction The type of action being performed (LOGIN, SIGNUP, etc.). Default is [RecaptchaAction.LOGIN]
+ * @property timeoutInMills Timeout for ReCaptcha verification in milliseconds. Default is 10000L
+ * @property customPayload Optional custom JSON payload to be sent with the verification request
+ * @property logger Logger instance for logging verification process. Default is [Logger.Companion.WARN]
+ */
+@PingDsl
+class ReCaptchaEnterpriseConfig {
+
+    /** The type of action being performed (LOGIN, SIGNUP, etc.) */
+    var recaptchaAction: RecaptchaAction = RecaptchaAction.LOGIN
+
+    /** Timeout for ReCaptcha verification in milliseconds */
+    var timeoutInMills: Long = 10000L
+
+    /** Custom payload to be sent with the verification request */
+    var customPayload: JsonObject? = null
+
+    /** Logger instance for logging verification process */
+    var logger: Logger = Logger.WARN
+}
+
+/**
+ * Journey callback for handling ReCaptcha Enterprise verification challenges.
+ *
+ * This callback integrates Google's ReCaptcha Enterprise service into Journey
+ * authentication flows. It handles the verification process by:
+ * - Receiving site key configuration from the server
+ * - Creating ReCaptcha clients using the configured provider
+ * - Executing verification actions with proper timeout handling
+ * - Returning verification tokens for server validation
+ *
+ * The callback supports customizable actions, timeouts, and client providers
+ * to accommodate different verification requirements and testing scenarios.
+ *
+ * Example usage:
+ * ```kotlin
+ * val callback = journey.next() as ReCaptchaEnterpriseCallback
+ * val result = callback.verify {
+ *     recaptchaAction = RecaptchaAction.LOGIN
+ *     timeoutInMills = 15000L
+ * }
+ * ```
+ *
+ * @property reCaptchaSiteKey The ReCaptcha site key received from the server configuration
+ * @property customPayload Optional custom JSON payload to be sent with the verification request
+ * @property logger Logger instance for logging verification process
+ */
+class ReCaptchaEnterpriseCallback : AbstractCallback() {
+    /**
+     * The ReCaptcha site key received from the server configuration.
+     * This key is automatically populated during callback initialization.
+     */
+    var reCaptchaSiteKey: String = ""
+        private set
+
+    var customPayload: JsonObject? = null
```

Comment:
I think we can remove this attribute and just provide one way to set the customPayload, using the ReCaptchaEnterpriseConfig